soccer/ui/MainWindow.cpp

```diff
@@ -544,234 +536,290 @@ void MainWindow::updateViews() {
-                    break;
-                case RJ2015:
-                    robotModel = "RJ2015";
-                    break;
-                default:
-                    robotModel = "Unknown Bot";
+    if (currentFrame != nullptr) {
+        // update robot status list
+        for (int shell = 0; shell < Num_Shells; shell++) {
+            // Search for the corresponding references.
+            auto maybe_rx =
+                [&]() -> std::optional<
+                          std::reference_wrapper<const Packet::RadioRx>> {
+                for (int i = 0; i < currentFrame->radio_rx_size(); i++) {
+                    if (currentFrame->radio_rx(i).robot_id() == shell) {
+                        return currentFrame->radio_rx(i);
+                    }
+                }
+                return std::nullopt;
+            }();
+            auto maybe_robot = [&]()
+                -> std::optional<
+                    std::reference_wrapper<const Packet::LogFrame_Robot>> {
+                for (int i = 0; i < currentFrame->self_size(); i++) {
+                    if (currentFrame->self(i).shell() == shell) {
+                        return currentFrame->self(i);
+                    }
+                }
+                return std::nullopt;
+            }();
+
+            // If rx is missing, skip this one.
+            if (!maybe_rx.has_value()) {
+                continue;
             }
-            statusWidget->setRobotModel(robotModel);
 
-// uncomment this #define to test the display of a variety of
-// different errors #define DEMO_ROBOT_STATUS
+            const auto& rx = maybe_rx.value().get();
+
+            // a robot shows up in the status list if it's reachable via radio
+            bool shouldDisplay =
+                currentFrame->timestamp() - rx.timestamp() < 5000000;
+
+            // see if it's already in the robot status list widget
+            bool displaying =
+                _robotStatusItemMap.find(shell) != _robotStatusItemMap.end();
+
+            if (shouldDisplay && !displaying) {
+                // add a widget to the list for this robot
+
+                auto item_owned = std::make_unique<QListWidgetItem>();
+                auto* item = item_owned.get();
+                _robotStatusItemMap[shell] = std::move(item_owned);
+                _ui.robotStatusList->addItem(item);
+
+                // The item's widget is managed by Qt
+                // (setItemWidget takes ownership).
+                auto* statusWidget = new RobotStatusWidget();  // NOLINT
+                item->setSizeHint(statusWidget->minimumSizeHint());
+                _ui.robotStatusList->setItemWidget(item, statusWidget);
+
+                // set shell ID
+                statusWidget->setShellID(shell);
+
+                // set team
+                statusWidget->setBlueTeam(
+                    _context->game_settings.requestBlueTeam);
+
+                // TODO(Kyle): set board ID
```

Comment:
Make an issue for it and include the issue number?

Reply:
We actually don't have board IDs anymore. I'm going to do one more pass through this and clean up the variables that don't apply anymore from the log frame, considering this is already sort of breaking backwards compatibility anyway I'm okay with that.